app/models/user.rb

```diff
@@ -11,4 +11,93 @@ class User < ApplicationRecord
   has_one :images, as: :imageable, dependent: :destroy
 
   enum role: {user: 0, admin: 1}, _prefix: true
+
+  EMAIL_REGEX = Settings.regex.email.freeze
+
+  scope :activated, ->{where activated: true}
+
+  attr_accessor :remember_token, :activation_token, :reset_token
+
+  validates :name, presence: true, length: {maximum: Settings.digits.digit_50}
+  validates :email, presence: true,
+            length: {maximum: Settings.digits.digit_255},
+            format: {with: EMAIL_REGEX}, uniqueness: true
+  validates :password_digest, presence: true,
```

Comment:
```suggestion
  validates :password, presence: true,
```
Sao lại validate password_digest anh nhỉ? Mình nên validate những thứ mà người dùng input chứ nhỉ?

Reply:
updated